### PR TITLE
Config file for sphinx-documentation: Improve comments and fix typos

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Just a few typo fixes in the config file for sphinx documentation.

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -136,17 +136,17 @@
...
-    # Sort gallery example by file name instead of number of lines (default)
+    # Sort gallery examples by file name instead of number of lines (default)
     "within_subsection_order": ExampleTitleSortKey,
-    # directory where function granular galleries are stored
+    # Directory where function granular galleries are stored
     "backreferences_dir": "api/generated/backreferences",
-    # Modules for which function level galleries are created.  In
-    # this case sphinx_gallery and numpy in a tuple of strings.
+    # Modules for which function level galleries are created. In
+    # this case sphinx_gallery and numpy are a tuple of strings.
```
Comment (yvonnefroehlich) [inline doc/conf.py:144]: Actually this comment appears a bit strange / unclear to me here.

Comment (seisman) [reply doc/conf.py:144]: xref: https://sphinx-gallery.github.io/stable/configuration.html#examples-using-numpy-exp

Comment (seisman) [inline doc/conf.py:144]:
```suggestion
    # Modules for which function level galleries are created.
```

File: doc/conf.py [hunk 2]
```diff
@@ -136,17 +136,17 @@
...
     "doc_module": "pygmt",
```
Comment (seisman) [inline doc/conf.py:145]:
```suggestion
    "doc_module": ("pygmt",),
```

File: doc/conf.py [hunk 3]
```diff
@@ -136,17 +136,17 @@
...
     # Insert links to documentation of objects in the examples
     "reference_url": {"pygmt": None},
     "image_scrapers": (PyGMTScraper(),),
-    # Removes configuration comments from scripts
+    # Remove configuration comments from scripts
     "remove_config_comments": True,
     # Disable "nested_sections" (default is True), to
```
Comment (seisman) [inline doc/conf.py:151]: Please also rewrap the lines 151-153 to 88 characters.

Comment (yvonnefroehlich) [reply doc/conf.py]: Done in PR #3573.

File: doc/conf.py (hunk 1 above)
Comment (yvonnefroehlich) [reply doc/conf.py]: Reformulated in commit https://github.com/GenericMappingTools/pygmt/pull/3573/commits/845aa96af7229c5911559d3008d6ef0744245dbd.

File: doc/conf.py (hunk 2 above)
Comment (yvonnefroehlich) [reply doc/conf.py]: Included in commit https://github.com/GenericMappingTools/pygmt/pull/3573/commits/845aa96af7229c5911559d3008d6ef0744245dbd.

**Comment by yvonnefroehlich** [comment]
New PR at
- #3573 .

**Pull request closed without merging**